austation/code/game/materialization.dm

```diff
@@ -0,0 +1,297 @@
+/obj/structure/table/mat_shiritori
+	name = "\improper Materialization Shiritori"
+	desc = "A magical Shiritori table capable of temporarily materializing any object that exists in this reality."
+	icon = 'austation/icons/obj/structures/magic_table.dmi'
+	icon_state = "magic_table"
+	max_integrity = 300
+	canSmoothWith = null
+	flags_1 = NODECONSTRUCT_1
+	var/current_letter = ""
+	var/time_left = 30 SECONDS
+	var/active = FALSE
+	var/game_starting = FALSE // prep period
+	var/start_time = 10 // how many seconds people get to opt in
+	var/turns = 0 // how many turns have passed
+	var/static/list/atom_list // starts off as empty, we only want to make this if we have a table, otherwise we're just wasting memory
+	var/static/instances = 0 // amount of tables that currently exist. Used for clearing atom list
+	var/list/spent_objs = list() // typepaths that have been used before
+	var/list/entities = list() // spawned entities that are currently in existance
+	var/list/players = list()
+	var/mob/living/carbon/human/current_player
+	var/list/knockouts = list()
+	var/static/list/blacklist // handled in setup_blacklist(). Contains root paths that aren't supposed to be spawned (Yes, a bit of a pain to maintain, but it's not the end of the world if someone forgets to update it. The alternative is having a variable on every damn datum)
+
+/obj/structure/table/mat_shiritori/Destroy()
+	if(active)
+		end_game()
+		return ..()
+	STOP_PROCESSING(SSobj, src)
+	QDEL_LIST(entities)
+	current_player = null
+	players = null
+	knockouts = null
+	return ..()
+
+/obj/structure/table/mat_shiritori/proc/ready_up()
+	if(active || game_starting)
+		return
+	instances++
+	setup_blacklist()
+	players.len = 0
+	knockouts.len = 0
+	time_left = initial(time_left)
+	say("Game starting in [start_time] seconds. Place your hand on the table to play.")
+	INVOKE_ASYNC(src, .proc/start_game)
+
+//  Called after everything is ready
+/obj/structure/table/mat_shiritori/proc/start_game()
+	game_starting = TRUE
+	for(var/i in 1 to start_time)
+		var/countdown = start_time - (i - 1)
+		switch(countdown)
+			if(5)
+				say("Game starting in 5...")
+			if(1 to 4)
+				say("[countdown]...")
+		sleep(10)
+	for(var/mob/living/L as() in players)
+		if(QDELETED(L) || L.stat == DEAD)
+			remove_player(L)
+			continue
+	if(length(players) > 1)
+		if(!atom_list)
+			atom_list = generate_name_list(typesof(/atom/movable))
+		active = TRUE
+		current_player = pick(players)
+		say("[current_player] will start.")
+		to_chat(current_player, "<span class='notice'>It is now your turn.</span>")
+		playsound(src, 'sound/effects/gong.ogg', 100, 0, 5)
+		START_PROCESSING(SSobj, src)
+	else
+		say("Insufficient players")
```

Comment:
Pacifism isn't getting removed if the game fails to start, nor does the players list get cleared
```suggestion
		say("Insufficient players")
		end_game()
```